tests/unit/test_charm.py

```diff
@@ -34,10 +34,27 @@ def setUp(self, *unused):
     def test_grafana_is_provided_port_and_source(self, *unused):
         rel_id = self.harness.add_relation("grafana-source", "grafana")
         self.harness.add_relation_unit(rel_id, "grafana/0")
+        fqdn = socket.getfqdn()
         grafana_host = self.harness.get_relation_data(rel_id, self.harness.model.unit.name)[
             "grafana_source_host"
         ]
-        self.assertEqual(grafana_host, "{}:{}".format("1.1.1.1", "9090"))
+        self.assertEqual(grafana_host, "http://{}:{}".format(fqdn, "9090"))
+
+    @patch_network_get(private_address="1.1.1.1")
+    def test_web_external_url_is_passed_to_grafana(self, *unused):
+        self.harness.set_leader(True)
+        self.harness.update_config({"web_external_url": "http://test:80/foo/bar"})
```

Comment:
The `web_external_url` option overrides whatever may come from Traefik.

We must add a test with the URL coming from Traefik.

Reply:
I tried two hours to debug SDI to make one such test. I give up :D